### ISSUE TITLE
An Issue serializing Structures with foreign keys inherited from included structure 

### ISSUE THREAD

**Issue by LSabap** [abapGit/abapGit#4897]
Hello,
we have faced with an issue that after pull a structure with foreign keys inherited from included structure we lost these definitions (DB Table DD08L - Original records contain value in the attribute NOINHERIT = 'N'). The records with NOINHERIT = 'N' are deleted in the method ZIF_ABAPGIT_OBJECT~SERIALIZE of the class ZCL_ABAPGIT_OBJECT_TABL in the section 
"* remove foreign keys inherited from .INCLUDEs". 
After a small adjustment (comment the line "DELETE lt_dd08v WHERE noinherit = 'N'.") the pull finished proper and all records from DD08L have been pulled. We would like ask what is reason of this section or whether it is a bug. 

Many thanks for you support/help.
Libor

(Left side: src from source system, right side: info about no existing keys)
  
![DD08L_Issue](https://user-images.githubusercontent.com/89007146/129529669-cf00bcf1-4ce1-465f-9d30-8173cf1eebb1.jpg)

**Comment by mbtools** [comment]
https://github.com/abapGit/abapGit/blob/main/src/objects/zcl_abapgit_object_tabl.clas.abap#L1009-L1018

The change goes back 5y
https://github.com/abapGit/abapGit/commit/17cf0a7b29df680918591f2126000543ca23cd96

Have a look at https://github.com/abapGit/abapGit/issues/4865 first (diffs due to different releases). Are your source and target systems on the same release?

**Comment by LSabap** [comment]
Yes, we have two different systems: SAP S/4 HANA (S4CORE 107) and SAP S/4 HANA Cloud (SAP_CLOUD 786).
But I guess the issue isn't caused by different systems because we reproduced it only on one system ( SAP S/4 HANA Cloud ) as well.
We staged (method ZIF_ABAPGIT_OBJECT~SERIALIZE) the structure with include to GitHub and we could see foreign keys from included include are removed here. As next step we changed the structure definition (we only replaced one DTEL with another DTEL) in the system (SE11) and pulled the structure back from GitHub into system. And then we lost foreign keys definitions as result. It means the issue isn't depend on the system.

Many thanks
